src/ui/training/index.ts

```diff
@@ -7,22 +7,21 @@ import { emptyFen } from '../../utils/fen'
 import * as helper from '../helper'
 import layout from '../layout'
 import ViewOnlyBoard from '../shared/ViewOnlyBoard'
-import { connectingHeader } from '../shared/common'
-
+import { syncPuzzles } from './utils'
 import { renderContent, renderHeader, overlay } from './trainingView'
 import * as xhr from './xhr'
 import TrainingCtrl from './TrainingCtrl'
+import { connectingHeader } from '../shared/common'
+import { loadOfflinePuzzle, puzzleLoadFailure } from './utils'
+import { State } from './interfaces'
+import { PuzzleData } from '../../lichess/interfaces/training'
 
 interface Attrs {
   id?: string
   initFen?: string
   initColor?: Color
 }
 
-interface State {
```

Comment:
Leave interface State here, it's done like this in the rest of the app. It's more convenient to see what is the component interface directly where it's defined.